src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -139,12 +139,12 @@ end
 
   mag_en = 0.5*(B1^2 + B2^2 + B3^2)
 
-  if orientation == 1
-    f_B1 = 0.0
-    f_B2 = v1_plus * B2 - v2_plus * B1
-    f_B3 = v1_plus * B3 - v3_plus * B1
+  f = zeros(MVector{nvariables(equations), eltype(u)})
 
-    f = (f_B1, f_B2, f_B3)
+  if orientation == 1
+    f[1] = 0.0
```

Comment:
```suggestion
    f[1] = 0
```
This should avoid possible conversion issues when we use other types

Reply:
In this case I suggest to apply this consistently everywhere. Right now, there are many places where `zero(eltype(u))` is used - I think for better readability and consistency, these should be changed to `0` as well where possible.

Reply:
Done in 179877b